web/crux/src/domain/container-log-stream.ts

```diff
@@ -0,0 +1,89 @@
+import { finalize, Observable, startWith, Subject } from 'rxjs'
+import { PreconditionFailedException } from 'src/exception/errors'
+import { AgentCommand } from 'src/grpc/protobuf/proto/agent'
+import { ContainerIdentifier, ContainerLogMessage } from 'src/grpc/protobuf/proto/common'
+import GrpcNodeConnection from 'src/shared/grpc-node-connection'
+
+export type ContainerLogStreamCompleter = Subject<unknown>
+
+export default class ContainerLogStream {
+  private stream = new Subject<ContainerLogMessage>()
+
+  private started = false
+
+  private completer: ContainerLogStreamCompleter = null
+
+  constructor(private id: string | undefined, private prefixName: ContainerIdentifier | undefined) {}
+
+  start(commandChannel: Subject<AgentCommand>) {
+    if (this.started) {
+      return
+    }
+
+    // TODO(robot9706): tail const
```

Comment:
If you want to add a const for 40, please do it. Otherwise, create a JIRA ticket and add the number to the note, please.

Reply:
I meant that for now it's a const value, and we could move it to UI in the future. I can move it to a const though.